src/otx/core/model/entity/anomaly.py

```diff
@@ -0,0 +1,257 @@
+"""Base Anomaly OTX model."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import onnx
+import openvino
+import torch
+from anomalib import TaskType
+from torch import nn
+
+from otx.core.data.dataset.base import LabelInfo
+from otx.core.data.entity.anomaly import (
+    AnomalyClassificationBatchPrediction,
+    AnomalyDetectionDataBatch,
+    AnomalySegmentationDataBatch,
+)
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.model.entity.base import OTXModel, OVModel
+from otx.core.types.precision import OTXPrecisionType
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from openvino.model_api.models import Model
+    from openvino.model_api.models.anomaly import AnomalyResult
+    from torchvision.transforms.v2 import Transform
+
+
+class _AnomalibLightningArgsCache:
+    """Caches args for the anomalib lightning module.
+
+    This is needed as the arguments are passed to the OTX model. These are saved and used by the OTX anomaly
+    lightning model.
+    """
+
+    def __init__(self):
+        self._args: dict[str, Any] = {}
+
+    def update(self, **kwargs) -> None:
+        """Add args to cache."""
+        self._args.update(kwargs)
+
+    def get(self) -> dict[str, Any]:
+        """Get cached args."""
+        return self._args
+
+
+@dataclass
+class _OVModelInfo:
+    """OpenVINO model information."""
+
+    image_shape: Sequence[int] = (256, 256)
+    image_threshold: float = 0.5
+    pixel_threshold: float = 0.5
+    task: TaskType = TaskType.CLASSIFICATION
+    mean_values: Sequence[float] = (0.0, 0.0, 0.0)
+    scale_values: Sequence[float] = (1.0, 1.0, 1.0)
+    normalization_scale: float = 1.0
+    orig_height: int = 256
+    orig_width: int = 256
+
+
+class _AnomalyModelExporter(OTXModelExporter):
+    def __init__(
+        self,
+        model_info: _OVModelInfo,
+    ) -> None:
+        self.model_info = model_info
+        metadata = {
+            ("model_info", "image_threshold"): model_info.image_threshold,
+            ("model_info", "pixel_threshold"): model_info.pixel_threshold,
+            ("model_info", "normalization_scale"): model_info.normalization_scale,
+            ("model_info", "orig_height"): model_info.orig_height,
+            ("model_info", "orig_width"): model_info.orig_width,
+            ("model_info", "image_shape"): model_info.image_shape,
+            ("model_info", "labels"): "Normal Anomaly",
+            ("model_info", "model_type"): "AnomalyDetection",
+            ("model_info", "task"): model_info.task.value,
+        }
+        super().__init__(
+            input_size=(1, 3, *model_info.image_shape),
+            mean=model_info.mean_values,
+            std=model_info.scale_values,
+            swap_rgb=True,  # set reverse input channels to True
```

Comment:
Is this because anomaly model requires BGR as an input?

Reply:
In this case I would like the inputs to be in `RGB` format. If I understood the doc string in the base class correctly `(Whether to convert the image from BGR to RGB Defaults to False.)` then I should set it to True to get RGB inputs to the model. Or did I misunderstand something?